uaclient/cli.py

```diff
@@ -297,6 +315,48 @@ def action_detach(args, cfg):
     return 0
 
 
+def action_attach_premium(args, cfg):
+    from uaclient.clouds import identity
+
+    if cfg.is_attached:
+        print(
+            "This machine is already attached to '{}'.".format(
+                cfg.accounts[0]["name"]
+            )
+        )
+        return 0
+    if os.getuid() != 0:
+        raise exceptions.NonRootUserError()
+    cloud_type = identity.get_cloud_type()
+    if cloud_type not in ("aws", "ec2"):
+        print(
+            "Premium image support is not supported on '%s'.\n"
+            "For more information see: https://ubuntu.com/advantage"
+            % cloud_type
+        )
+        return 0
+    instance = identity.cloud_instance_factory()
+    contract_client = contract.UAContractClient(cfg)
+    pkcs7 = instance.identity_doc
+    contractTokenResponse = contract_client.request_premium_aws_attach(pkcs7)
```

Comment:
It seems strange to have some of the cloud-specific logic captured in the `UAPremiumCloudInstance` sub-classes but still have AWS hard-coded here. We don't necessarily need to address this before we land it (because we're only going to support AWS to start with), but I would like us to be sure we know how we're going to encapsulate this better in future, to be sure we aren't heading into a dead end.